geode-core/src/upgradeTest/java/org/apache/geode/security/AuthExpirationDUnitTest.java

```diff
@@ -109,4 +114,82 @@ public void clientShouldReAuthenticateWhenCredentialExpiredAndOperationSucceed()
     assertThat(region.size()).isEqualTo(2);
   }
 
+  @Test
+  public void userShouldReAuthenticateWhenCredentialExpiredAndOperationSucceed() throws Exception {
+    int serverPort = server.getPort();
+    ClientVM clientVM = lsRule.startClientVM(0, clientVersion,
+        c -> c.withMultiUser(true)
+            .withProperty(SECURITY_CLIENT_AUTH_INIT, UpdatableUserAuthInitialize.class.getName())
+            .withPoolSubscription(true)
+            .withServerConnection(serverPort));
+
+    clientVM.invoke(() -> {
+      UpdatableUserAuthInitialize.setUser("serviceUser0");
+      UpdatableUserAuthInitialize.setAttempts(0);
+      UpdatableUserAuthInitialize.setAlternate(false);
+      ClientCache clientCache = ClusterStartupRule.getClientCache();
+      assert clientCache != null;
+      clientCache.createClientRegionFactory(ClientRegionShortcut.PROXY).create("region");
+      Properties userSecurityProperties = new Properties();
+      userSecurityProperties.put("security-username", "serviceUser0");
+      userSecurityProperties.put("security-password", "serviceUser0");
+      RegionService regionService = clientCache.createAuthenticatedView(userSecurityProperties);
+      Region<Object, Object> region = regionService.getRegion("/region");
+      region.put(0, "value0");
+      regionService.close();
+    });
+
```

Comment:
Why are splitting this blocks for the same VM?

Reply:
> I am confused about the repetition of the `clientvm blocks rather than continuing it.

You're right, in most cases they can be continued. I organized it this way as logical steps but unless we add a user to the expired user list in between there is no need for separation of the blocks. I will change it.

Reply:
One note that is most likely irrelevant in this case is that splitting it takes more time for context switching.